packages/app/src/background/services/credentials/index.ts

```diff
@@ -43,6 +48,26 @@ export default class VerifiableCredentialsService implements IBackupable {
     return VerifiableCredentialsService.INSTANCE;
   }
 
+  addVerifiableCredentialRequest = async (serializedVerifiableCredential: string): Promise<void> => {
+    await deserializeVerifiableCredential(serializedVerifiableCredential);
```

Comment:
I guess you meant here to validate first before executing `ADD_VERIFIABLE_CREDENTIAL`. It better to have a validate`validateVerifiableCredential` independent function to be executed as a middleware step before executing `addVerifiableCredentialRequest`. 

```ts
    this.handler.add(
      RPCAction.ADD_VERIFIABLE_CREDENTIAL_REQUEST,
      this.lockService.ensure,
      validateVerifiableCredential,
      this.verifiableCredentialsService.addVerifiableCredentialRequest,
    );
```

That means `validateVerifiableCredential` should be able to accept `serializedVerifiableCredential`, then it calls `deserializeVerifiableCredential` to deserialize.